contracts/LiquidityBridgeContract.sol

```diff
@@ -225,69 +258,86 @@ contract LiquidityBridgeContract {
         }
-
-        if (transferredAmount == -200 || transferredAmount == -100) {
+				
+        if (transferredAmountOrErrorCode == BRIDGE_REFUNDED_LP_ERROR_CODE || transferredAmountOrErrorCode == BRIDGE_REFUNDED_USER_ERROR_CODE) {
             // Bridge cap exceeded
-            callRegistry[quoteHash].timestamp = 0;
-            callRegistry[quoteHash].success = false;
-            emit BridgeCapExceeded(quoteHash, transferredAmount);
-            return transferredAmount;
-        }
-
-        if (transferredAmount > 0 && callRegistry[quoteHash].timestamp > 0) {
+            processedQuotes[quoteHash] = PROCESSED_QUOTE_CODE;
+            delete callRegistry[quoteHash];
+            emit BridgeCapExceeded(quoteHash, transferredAmountOrErrorCode);
+            return transferredAmountOrErrorCode;
+        }        
+	
+        if (callRegistry[quoteHash].timestamp > 0) {
             uint refundAmount;
 
             if (callRegistry[quoteHash].success) {
-                refundAmount = min(uint(transferredAmount), quote.value + quote.callFee);
-                callRegistry[quoteHash].success = false;
+                refundAmount = min(uint(transferredAmountOrErrorCode), quote.value + quote.callFee);
             } else {
-                refundAmount = min(uint(transferredAmount), quote.callFee);
+                refundAmount = min(uint(transferredAmountOrErrorCode), quote.callFee);
             }
             increaseBalance(quote.liquidityProviderRskAddress, refundAmount);
-            int256 remainingAmount = transferredAmount - int(refundAmount);
+            int256 remainingAmount = transferredAmountOrErrorCode - int(refundAmount);
             
-            if (remainingAmount > 0) {
-                (bool success, ) = quote.rskRefundAddress.call{value : uint(remainingAmount)}("");
-                require(success, "Refund failed");
+            if (remainingAmount > dust) {
+                quote.rskRefundAddress.transfer(uint(remainingAmount));
                 emit Refund(quote.rskRefundAddress, remainingAmount, quoteHash);
             }            
-            callRegistry[quoteHash].timestamp = 0;
-        } else if (transferredAmount > 0) {
-            (bool success, ) = quote.rskRefundAddress.call{value : uint256(transferredAmount)}("");
-            require(success, "Refund failed");
-            emit Refund(quote.rskRefundAddress, transferredAmount, quoteHash);
+        } else {
+            int256 refundAmount = transferredAmountOrErrorCode;
+
+            if (performCall && refundAmount >= int64(quote.value)) {
+                (bool callSuccess, ) = quote.contractAddress.call{gas:quote.gasLimit, value: quote.value}(quote.data);
+                emit CallForUser(msg.sender, quote.contractAddress, quote.gasLimit, quote.value, quote.data, callSuccess, quoteHash);
+
+                if (callSuccess) {
+                    refundAmount -= int64(quote.value);
```

Comment:
Will there be a way to withdraw in the future the sum of all dusts ?
Should we setup a notaryAccount in the contract that is able to withdraw any balance that is stuck on the contract and does not belong to anybody ? I think we should, and also allow this notaryAddress to be changed (only the notary can change it), so we can set it to zero if we see there is no need for it.

Reply:
I also think the notaryAddress should be able to move tokens stuck into this contract. It's common for users to mistake addresses and send tokens where they should not. If we don't give a failback method for someone to recover those tokens, they will be lost.

Reply:
Ok, but I would leave this for a different PR

Reply:
Another option would be to give this to the LP by increasing its balance

Reply:
I'm not sure. We can do that for now. Let's leave this for another PR.